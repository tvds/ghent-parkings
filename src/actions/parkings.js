// SET_PARKINGS
export const setParkings = (parkings) => ({
  type:'SET_PARKINGS',
  parkings
});

export const startSetParkings = () => {
  return (dispatch) => {  
    return fetch("https://datatank.stad.gent/4/mobiliteit/bezettingparkingsrealtime.json")
    .then(results =>  results.json())
    .then(data => {
      dispatch(setParkings(data))
    }).catch(error => console.log(error));
  };
}

