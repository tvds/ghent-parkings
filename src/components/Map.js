import React from "react";
import {
  withGoogleMap,
  GoogleMap,
  withScriptjs,
  Marker,
  InfoWindow
} from "react-google-maps";
import { compose, withProps, withStateHandlers } from "recompose";

const MapWithPlaces = compose(
  withProps({
    googleMapURL:
      "https://maps.googleapis.com/maps/api/js?key=AIzaSyDPhmjwwm96eMl8pDEO5MJMDmPshhlGfyg&libraries=geometry,drawing,places",
    loadingElement: <div className="loadingElement" />,
    containerElement: <div className="containerElement" />,
    mapElement: <div className="mapElement" />
  }),
  withStateHandlers(
    props => ({
      infoWindows: props.places.map(p => {
        return { isOpen: false };
      })
    }),
    {
      onToggleOpen: ({ infoWindows }) => selectedIndex => ({
        infoWindows: infoWindows.map((iw, i) => {
          iw.isOpen = selectedIndex === i;
          return iw;
        })
      })
    }
  ),
  withScriptjs,
  withGoogleMap
)(props => (
  <GoogleMap defaultZoom={props.zoom} defaultCenter={props.center} >
    {props.places &&
      props.places.map((place, i) => {
        let lat = parseFloat(place.latitude, 10);
        let lng = parseFloat(place.longitude, 10);

        return (
          <Marker
            id={place.id}
            key={place.id}
            label={i + 1 + ''}
            position={{ lat: lat, lng: lng }}
            title="Click to zoom"
            onClick={props.onToggleOpen.bind(this, i)}
          >
            {props.infoWindows[i].isOpen && (
              <InfoWindow onCloseClick={props.onToggleOpen.bind(i)}>
                <div>{place.description}</div>
              </InfoWindow>
            )}
          </Marker>
        );
      })}
  </GoogleMap>
));

export default MapWithPlaces;
