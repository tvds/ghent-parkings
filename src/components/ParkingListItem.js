import React from 'react';
import { Link } from 'react-router-dom';

export class ParkingListItem extends React.Component {
  render() {
    return (
      <div className={this.props.active ? 'parked' : ''} >
        <Link className="list-item" to={`/details/${this.props.id}`}>
          <div>
            <h3 className="list-item__title">{this.props.description}</h3>
            {this.props.address} {this.props.active ? this.props.parkingStatus.availableCapacity + 1 : this.props.parkingStatus.availableCapacity}/{this.props.parkingStatus.totalCapacity}  Open:{this.props.parkingStatus.open + ""}
          </div>
        </Link>
        <button onClick={this.props.onPark}>Park</button>
      </div>
    );
  };

}

export default ParkingListItem;