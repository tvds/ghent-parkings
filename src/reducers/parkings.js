// Parkings Reducer

const parkingsReducerDefaultState = [];

export default (state = parkingsReducerDefaultState, action) => {
    switch(action.type) {
        case 'SET_PARKINGS':
            return action.parkings;
        default:
            return state;
    }
};