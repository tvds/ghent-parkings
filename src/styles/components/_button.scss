
.button {
  background-color: $blue;
  color: white;
  display:inline-block;
  padding: $s-size;
  border: none;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  text-decoration: none;
  border-radius: 3px;
  &:hover {
    background-color: lighten($blue, 20%);
  }
}

 .button--link {
  background: none;
 }

 .button--secondary {
   background: #888;
 }