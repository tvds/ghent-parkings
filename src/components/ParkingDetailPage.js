
import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Map from "./Map";

const ParkingDetailPage = (props) => {
  return (
    <div>
      <div className="detail-header">
        <Link className="detail-header__link" to="/">
          <div className="content-container">
            <FontAwesomeIcon icon={['fas', 'chevron-left']} />  Return to dashboard
          </div>
        </Link>
      </div>
      <div className="detail-body">
        <div className="content-container detail-body__data">
          <h1 className="detail-body__title">
            {props.parking.description}
          </h1>
          <div className="detail-body__sub-title">{props.parking.contactInfo}</div>
          <h3>Opening hours</h3>
          {
            props.parking.openingTimes[0].days.map((day, key) => {
              return <div key={key}>{day}: {props.parking.openingTimes[0].from} - {props.parking.openingTimes[0].to}</div>;
            })
          }
        </div>
        <Map
          center={{ lat: 51.052757, lng: 3.718902 }}
          zoom={13}
          places={[props.parking]}
        />
      </div>
    </div>
  );
}


const mapStateToProp = (state, props) => {
  return {
    parking: state.parkings.find((parking) => parking.id.toString() === props.match.params.id)
  };
};

export default connect(mapStateToProp)(ParkingDetailPage);