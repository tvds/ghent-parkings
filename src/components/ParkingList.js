import React from 'react';
import ParkingListItem from './ParkingListItem';
import { connect } from 'react-redux';


const ParkingList = (props) => {
  return (
    <div className="content-container">
      <div className="list-header">
        <div>ParkingList</div>
      </div>
      <div className="list-body">
        {
          props.parkings.map((parking) => {
            return <ParkingListItem key={parking.id} {...parking} />;
          })
        }
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    parkings: state.parkings
  };
};

export default connect(mapStateToProps)(ParkingList);