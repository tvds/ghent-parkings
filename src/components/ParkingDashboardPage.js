import React from 'react';
import { connect } from 'react-redux';
import ParkingList from './ParkingList';
import Map from "./Map";

export class ParkingDashboardPage extends React.Component {
  render() {
    return (
      <div>
        <div className='dashboard-left'>
          <ParkingList parkings={this.props.parkings} />
        </div>
        <div className='dashboard-right'>
          <Map
            center={{ lat: 51.052757, lng: 3.718902 }}
            zoom={13}
            places={this.props.parkings}
          />
        </div>
      </div>
    );
  };
}

const mapStateToProps = (state) => {
  return {
    parkings: state.parkings
  };
};

export default connect(mapStateToProps)(ParkingDashboardPage);
