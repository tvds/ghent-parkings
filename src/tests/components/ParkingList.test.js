import React from 'react';
import { shallow } from 'enzyme';
import { ParkingList } from '../../components/ParkingList';
import parkings from '../fixtures/parkings';

test('should render ParkingList with parkings', () => {
  const wrapper = shallow(<ParkingList parkings={parkings} />);
  expect(wrapper).toMatchSnapshot();
});

