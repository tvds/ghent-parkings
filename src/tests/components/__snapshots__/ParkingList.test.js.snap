// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ParkingList with parkings 1`] = `
<div>
  <div
    className="list-body"
  >
    <ParkingListItem
      active={false}
      address="Sint-Michielsplein 8
    9000 Gent"
      capacityRounding={1}
      city={
        Object {
          "id": 1004,
          "name": "Gent",
        }
      }
      contactInfo="Tel.: 09 266 29 20"
      description="Sint-Michiels"
      id={18408}
      index={0}
      key="18408"
      lastModifiedDate="2016-04-12T12:58:15.673Z"
      latitude={51.05367}
      longitude={3.7186}
      name="P07 Sint-Michiels"
      onPark={[Function]}
      openingTimes={
        Array [
          Object {
            "days": Array [
              "MONDAY",
              "TUESDAY",
              "WEDNESDAY",
              "THURSDAY",
              "FRIDAY",
              "SATURDAY",
              "SUNDAY",
            ],
            "from": "00:00",
            "to": "23:59",
          },
        ]
      }
      parkingServer={
        Object {
          "id": 1005,
          "name": "ITG Gent",
        }
      }
      parkingStatus={
        Object {
          "activeRoute": "",
          "availableCapacity": 93,
          "lastModifiedDate": "03/03/2019 15:41:00",
          "open": true,
          "suggestedCapacity": "ACTUAL",
          "totalCapacity": 450,
        }
      }
      suggestedFreeThreshold={4}
      suggestedFullThreshold={4}
      totalCapacity={450}
    />
    <ParkingListItem
      active={false}
      address="Sint-Pietersplein 65
    9000 Gent"
      capacityRounding={1}
      city={
        Object {
          "id": 1004,
          "name": "Gent",
        }
      }
      contactInfo="Tel.: 09 266 29 60"
      description="Sint-Pietersplein"
      id={18411}
      index={1}
      key="18411"
      lastModifiedDate="2016-04-12T12:58:26.649Z"
      latitude={51.04171}
      longitude={3.72557}
      name="P10 Sint-Pietersplein"
      onPark={[Function]}
      openingTimes={
        Array [
          Object {
            "days": Array [
              "MONDAY",
              "TUESDAY",
              "WEDNESDAY",
              "THURSDAY",
              "FRIDAY",
              "SATURDAY",
              "SUNDAY",
            ],
            "from": "00:00",
            "to": "23:59",
          },
        ]
      }
      parkingServer={
        Object {
          "id": 1005,
          "name": "ITG Gent",
        }
      }
      parkingStatus={
        Object {
          "activeRoute": "",
          "availableCapacity": 449,
          "lastModifiedDate": "03/03/2019 15:41:00",
          "open": true,
          "suggestedCapacity": "ACTUAL",
          "totalCapacity": 708,
        }
      }
      suggestedFreeThreshold={3}
      suggestedFullThreshold={3}
      totalCapacity={708}
    />
    <ParkingListItem
      active={false}
      address="Vrijdagmarkt 1
    9000 Gent"
      capacityRounding={1}
      city={
        Object {
          "id": 1004,
          "name": "Gent",
        }
      }
      contactInfo="Tel.: 09 266 29 00
    (permanentie)
    Tel.: 09 266 29 01
    (tijdens kantooruren)"
      description="Vrijdagmarkt"
      id={18405}
      index={2}
      key="18405"
      lastModifiedDate="2016-04-12T12:57:26.709Z"
      latitude={51.05652}
      longitude={3.72595}
      name="P01 Vrijdagmarkt"
      onPark={[Function]}
      openingTimes={
        Array [
          Object {
            "days": Array [
              "MONDAY",
              "TUESDAY",
              "WEDNESDAY",
              "THURSDAY",
              "FRIDAY",
              "SATURDAY",
              "SUNDAY",
            ],
            "from": "00:00",
            "to": "23:59",
          },
        ]
      }
      parkingServer={
        Object {
          "id": 1005,
          "name": "ITG Gent",
        }
      }
      parkingStatus={
        Object {
          "activeRoute": "P01 hoofd aanrijroute",
          "availableCapacity": 47,
          "lastModifiedDate": "03/03/2019 15:43:00",
          "open": true,
          "suggestedCapacity": "ACTUAL",
          "totalCapacity": 647,
        }
      }
      suggestedFreeThreshold={3}
      suggestedFullThreshold={3}
      totalCapacity={647}
    />
    <ParkingListItem
      active={false}
      address="Savaanstraat 13
    9000 Gent"
      blurAvailability={false}
      capacityRounding={1}
      city={
        Object {
          "id": 1004,
          "name": "Gent",
        }
      }
      contactInfo="Tel.: 09 266 29 40"
      description="Savaanstraat"
      id={12687}
      index={3}
      key="12687"
      lastModifiedDate="2016-04-12T12:57:37.637Z"
      latitude={51.04862}
      longitude={3.72225}
      name="P04 Savaanstraat"
      onPark={[Function]}
      openingTimes={
        Array [
          Object {
            "days": Array [
              "MONDAY",
              "TUESDAY",
              "WEDNESDAY",
              "THURSDAY",
              "FRIDAY",
              "SATURDAY",
              "SUNDAY",
            ],
            "from": "00:00",
            "to": "23:59",
          },
        ]
      }
      parkingServer={
        Object {
          "id": 1005,
          "name": "ITG Gent",
        }
      }
      parkingStatus={
        Object {
          "activeRoute": "",
          "availableCapacity": 334,
          "lastModifiedDate": "03/03/2019 15:42:00",
          "open": true,
          "suggestedCapacity": "ACTUAL",
          "totalCapacity": 540,
        }
      }
      suggestedFreeThreshold={2}
      suggestedFullThreshold={2}
      totalCapacity={540}
    />
    <ParkingListItem
      active={false}
      address="Ramen 23
    9000 Gent"
      capacityRounding={1}
      city={
        Object {
          "id": 1004,
          "name": "Gent",
        }
      }
      contactInfo="Tel.: 09 266 29 70"
      description="Ramen"
      id={18414}
      index={4}
      key="18414"
      lastModifiedDate="2016-04-12T12:57:51.602Z"
      latitude={51.05532}
      longitude={3.71653}
      name="P08 Ramen"
      onPark={[Function]}
      openingTimes={
        Array [
          Object {
            "days": Array [
              "MONDAY",
              "TUESDAY",
              "WEDNESDAY",
              "THURSDAY",
              "FRIDAY",
              "SATURDAY",
              "SUNDAY",
            ],
            "from": "00:00",
            "to": "23:59",
          },
        ]
      }
      parkingServer={
        Object {
          "id": 1005,
          "name": "ITG Gent",
        }
      }
      parkingStatus={
        Object {
          "activeRoute": "",
          "availableCapacity": 84,
          "lastModifiedDate": "03/03/2019 15:43:00",
          "open": true,
          "suggestedCapacity": "ACTUAL",
          "totalCapacity": 266,
        }
      }
      suggestedFreeThreshold={8}
      suggestedFullThreshold={8}
      totalCapacity={266}
    />
    <ParkingListItem
      active={false}
      address="Seminariestraat 9
    9000 Gent"
      capacityRounding={1}
      city={
        Object {
          "id": 1004,
          "name": "Gent",
        }
      }
      contactInfo="Tel.: 09 266 29 50"
      description="Reep"
      id={18417}
      index={5}
      key="18417"
      lastModifiedDate="2016-04-12T12:58:03.931Z"
      latitude={51.05207}
      longitude={3.72981}
      name="P02 Reep"
      onPark={[Function]}
      openingTimes={
        Array [
          Object {
            "days": Array [
              "MONDAY",
              "TUESDAY",
              "WEDNESDAY",
              "THURSDAY",
              "FRIDAY",
              "SATURDAY",
              "SUNDAY",
            ],
            "from": "00:00",
            "to": "23:59",
          },
        ]
      }
      parkingServer={
        Object {
          "id": 1005,
          "name": "ITG Gent",
        }
      }
      parkingStatus={
        Object {
          "activeRoute": "",
          "availableCapacity": 225,
          "lastModifiedDate": "03/03/2019 15:42:00",
          "open": true,
          "suggestedCapacity": "ACTUAL",
          "totalCapacity": 470,
        }
      }
      suggestedFreeThreshold={5}
      suggestedFullThreshold={5}
      totalCapacity={470}
    />
  </div>
</div>
`;
